doc/playground.lua

```diff
@@ -0,0 +1,165 @@
+    bucket_count = 3000,
+    sharding = {
+        [box.info().cluster.uuid] = {
+            replicas = {
+                [box.info().uuid] = {
+                    uri = uri,
+                    name = 'storage',
+                    master = true,
+                },
+            },
+        },
+    },
+}
+vshard.storage.cfg(cfg, box.info().uuid)
+vshard.router.cfg(cfg)
+vshard.router.bootstrap()
+
+-- Create a space.
+box.once('developers', function()
+    box.schema.create_space('developers', {
+        format = {
+            {name = 'id', type = 'unsigned'},
+            {name = 'bucket_id', type = 'unsigned'},
+            {name = 'name', type = 'string'},
+            {name = 'surname', type = 'string'},
+            {name = 'age', type = 'number'},
+        }
+    })
+    box.space.developers:create_index('primary_index', {
+        parts = {
+            {field = 1, type = 'unsigned'},
+        },
+    })
+    box.space.developers:create_index('age_index', {
+        parts = {
+            {field = 5, type = 'number'},
+        },
+    })
+    box.space.developers:create_index('full_name', {
+        parts = {
+            {field = 3, type = 'string'},
+            {field = 4, type = 'string'},
+        },
+    })
+
+    -- Fill the space.
+    box.space.developers:insert({1, 7331, 'Alexey', 'Adams', 20})
+    box.space.developers:insert({2, 899, 'Sergey', 'Allred', 21})
+    box.space.developers:insert({3, 9661, 'Pavel', 'Adams', 27})
+    box.space.developers:insert({4, 501, 'Mikhail', 'Liston', 51})
+    box.space.developers:insert({5, 1993, 'Dmitry', 'Jacobi', 16})
+    box.space.developers:insert({6, 8765, 'Alexey', 'Sidorov', 31})
+end)
+
+-- Initialize crud.
+crud.init_storage()
+crud.init_router()
+
+-- {{{ help() and prev()
+
+local examples = {}
+local next_example = 1
+
+local function add_help(filename)
+    local fh = fio.open(filename)
+    local doc = fh:read()
+    fh:close()
+
+    local example = {}
+
+    local state = 'initial'
+    for _, line in ipairs(doc:split('\n')) do
+        if state == 'initial' then
+            if line:startswith('```') then
+                state = 'in_example'
+            end
+        elseif state == 'in_example' then
+            if line:startswith('```') then
+                table.insert(examples, table.concat(example, '\n'))
+                example = {}
+                state = 'initial'
+            else
+                table.insert(example, line)
+            end
+        else
+            assert(false)
+        end
+    end
+end
+
+local function help()
+    if next_example > #examples then
+        print('No more examples')
+        return
+    end
+
+    print(examples[next_example])
+    next_example = next_example + 1
+end
+
+local function prev()
+    if next_example - 2 < 1 then
+        print("You're on the first example")
+        return
+    end
+
+    print(examples[next_example - 2])
+    next_example = next_example - 1
+end
+
+add_help('doc/select.md')
```

Comment:
We need some test of the script on CI/CD. It's not a fact that developers will use it for each issue and the script can easily become outdated.

Reply:
Are you have some tests in the mind?

I can propose a smoke test: console is run, a simple select works, `help()` and `prev()` show something (w/o a check what exactly).

Reply:
It seems to me that some kind of unit testing will be over-engineering here, so we need to test an output.

If we can split Tarantool's log output and the script output then we can compare the output with a reference output (a ready file). It can be something like that:

- Some kind of 'help()/prev()/next()' sequence as input. Output is saved to a file.
- Compare the output file and the reference file.

At first glance, they should always be same. The examples themselves do not change often, so it should not be a big problem to update a reference output time to time.

Reply:
I would test the navigation on some testing samples. Duplicating of the documentation feels a bit too heavy here for me.

Reply:
`help()` / `prev()` part is not actual anymore, but the idea to test that the script is workeable still actual.

Reply:
Added a test.